bin/common.ml

```diff
@@ -1044,7 +1043,17 @@ let init ?action_runner ?log_file c =
     let+ w = Dune_rules.Workspace.workspace () in
     Dune_engine.Execution_parameters.builtin_default
     |> Dune_rules.Workspace.update_execution_parameters w);
-  Dune_rules.Global.init ~capture_outputs:c.builder.capture_outputs;
+  (* Setup color support *)
+  Dune_rules.Clflags.supports_color :=
+    c.builder.capture_outputs && Lazy.force Ansi_color.stderr_supports_color;
+  (* Setup initial environment *)
+  let env =
+    if !Dune_rules.Clflags.supports_color then
+      Dune_cli.Colors.setup_env_for_colors Env.initial
+    else Env.initial
```

Comment:
you can move most of this logic to dune_cli, and just expose the following function:

```
val color_support : capture_outputs:bool -> (Env.t * [`Color of bool])
```

Reply:
I'm not sure what you mean here. The reason I have to initialise supports_color is because of its use in context.ml. even if I refactor like you say I don't see the point. Perhaps I am missing something?